src/python/pants/backend/native/tasks/link_shared_libraries.py

```diff
@@ -167,12 +179,14 @@ def _execute_link_request(self, link_request):
     native_artifact = link_request.native_artifact
     output_dir = link_request.output_dir
     resulting_shared_lib_path = os.path.join(output_dir, native_artifact.as_shared_lib(platform))
+    self.context.log.debug("resulting_shared_lib_path: {}".format(resulting_shared_lib_path))
     # We are executing in the results_dir, so get absolute paths for everything.
+    buildroot = get_buildroot()
     cmd = ([linker.exe_filename] +
            self._get_shared_lib_cmdline_args(platform) +
-           link_request.external_libs_info.get_third_party_lib_args() +
-           ['-o', os.path.abspath(resulting_shared_lib_path)] +
-           [os.path.abspath(obj) for obj in object_files])
+           self._get_third_party_lib_args(link_request) +
+           ['-o', os.path.join(buildroot, resulting_shared_lib_path)] +
+           [os.path.join(buildroot, obj) for obj in object_files])
```

Comment:
is `obj` a path to somewhere? or do we have object files in the build root dir and why?

Reply:
Joining against the buildroot is necessary in compile to get correct paths to source files when we modify the buildroot (e.g. in our ctypes unit testing). I don't think this is correct for object files and other things stored in `.pants.d` and I will remove that now and I think it will work.

Reply:
👍

Reply:
Great catch, thanks a ton. Fixed in f86aedd!